tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,760 @@
+-uploadID    : comma-separated list of upload IDs (in table C<mri_upload>) to delete.
+
+
+=head1 DESCRIPTION
+
+This program deletes all the files and database records produced by the imaging pipeline for a given set
+of imaging uploads that have the same C<TarchiveID> in table C<mri_upload>. The script will issue and error
+message and exit if multiple upload IDs are passed on the command line and they do not all have the 
+same C<TarchiveID>. The script will remove the records associated to the imaging upload whose IDs are passed
+on the command line from the following tables: C<notification_spool>, C<tarchive_series>
+C<tarchive_files>, C<files_intermediary>, C<parameter_file>, C<files>, C<mri_violated_scans>
+C<mri_violations_log>, C<MRICandidateErrors>, C<mri_upload> and C<tarchive>. It will also delete from
+the file system the files that are associated to the upload and are listed in tables C<files>
+C<files_intermediary> and C<parameter_file>, along with the archive itself, whose path is stored in 
+table C<tarchive>. The script will abort and will not delete anything if there is QC information
+associated to the upload(s) (i.e entries in tables C<files_qcstatus> or C<feedback_mri_comments>).
+If the script finds a file that is listed in the database but that does not exist on the file system or
+is not readable, the script will issue an error message and abort, leaving the file system and database
+untouched. This behaviour can be changed with option C<-ignore>. By default, the script will create a
+backup of all the files that it plans to delete before actually deleting them. Use option C<-nobackup>
+to perform a 'hard' delete (i.e. no backup). The backup file name will be C<< mri_upload.<UPLOAD_ID>.tar.gz >>.
+Note that the file paths inside this backup archive are absolute.
+
+=head2 Methods
+
+=cut
+
+use strict;
+use warnings;
+
+use NeuroDB::DBI;
+use NeuroDB::ExitCodes;
+use NeuroDB::MRI;
+
+use Getopt::Tabular;       
+
+use File::Temp qw/tempfile/;
+
+use constant DEFAULT_PROFILE           => 'prod';
+use constant DEFAULT_DIE_ON_FILE_ERROR => 1;
+use constant DEFAULT_NO_BACKUP         => 0;
+
+use constant PIC_SUBDIR                => 'pic';
+
+my $profile        = DEFAULT_PROFILE;
+my $dieOnFileError = DEFAULT_DIE_ON_FILE_ERROR;
+my $noBackup       = DEFAULT_NO_BACKUP;
+my $uploadIDList   = undef;
+
+
+my @opt_table = (
+    ['-profile' , 'string'  , 1, \$profile, 
+     'name of config file in ../dicom-archive/.loris_mri (defaults to "prod")'],
+    ['-ignore'  , 'const'   , 0, \$dieOnFileError, 
+     'When performing the file backup, ignore files that do not exist or are not readable.'
+     . ' Default is to abort if such a file is found.'],
+    ['-nobackup', 'const'   , 0, \$noBackup,
+     'Do not backup anything. Default is to backup all files to be deleted '
+     . 'into an archive named "mri_upload.<uploadID>.tar.gz", in the '
+     . 'current directory'],
+    ['-uploadID', 'string', 1, \$uploadIDList,
```

Comment:
I would tend to clarify here that the list of provided uploadIDs must be associated to the same tarchiveID for the script to work. Cannot provide a list of uploadIDs that are associated to different tarchive. Otherwise, one could think that you could provide a list of uploadID to delete at once. That could be an interesting feature but not sure it would be easy to implement so maybe just clarify here that all specified uploadID in the command line should be associated to one and only one tarchiveID.

Reply:
Ok